src/power_grid_model/errors.py

```diff
@@ -109,6 +109,10 @@ class PowerGridDatasetError(PowerGridError):
     """Error occurs during dataset handling."""
 
 
+class NotImplementedError(PowerGridError):
+    """The functionality is either not supported or not yet implemented."""
```

Comment:
hmm, I see pylint doesn't like this (https://github.com/PowerGridModel/power-grid-model/actions/runs/11215571356/job/31173041487?pr=752)

```
 ************* Module power_grid_model.errors
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/power_grid_model/errors.py:112:0: W0622: Redefining built-in 'NotImplementedError' (redefined-builtin)
************* Module power_grid_model.core.error_handling
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/power_grid_model/core/error_handling.py:16:0: W0622: Redefining built-in 'NotImplementedError' (redefined-builtin)
```

maybe you can rename it to `PowerGridNotImplementedError` (cfr. `PowerGridUnreachableHitError` below).

As an alternative, we could also use the built-in `NotImplementedError`, with the danger that it does not derive from `PowerGridError`, which could surprise users. I don't think that that would be the way to go here.

We could let `PowerGridNotImplementedError` derive from both `PowerGridError` and `NotImplementedError`, but multiple-inheritance exception handling may also result in unexpected behavior

```py
>>> class A(Exception):
...   pass
...
>>> class B(A):         
...   pass
...
>>> class C(A, NotImplementedError):
...   pass
...
>>> try:        
...   raise B()
... except A:
...   print("A")
... except NotImplementedError:
...   print("NIE")
...
A
>>> try:         
...   raise C() 
... except A:    
...   print("A")
... except NotImplementedError:
...   print("NIE")
...
A
>>> try:
...   raise C()
... except NotImplementedError:
...   print("NIE")
... except A:    
...   print("A")
...
NIE
```